pkgs/servers/sks/default.nix

```
@@ -24,6 +24,10 @@ stdenv.mkDerivation rec {
     sed -i \
       -e "s:^LIBDB=.*$:LIBDB=-ldb:g" \
       Makefile.local
+
+    make cryptokit-1.7/README.txt
```

WOW... I've never seen an upstream hack like that...
https://github.com/SKS-Keyserver/sks-keyserver/blob/1.1.6/Makefile#L286

IMO we should document that part as the way this works is far from obvious, e.g.:
```suggestion
    # TODO: Drop after the next update (this patch is required for building with OCaml 4.04).
    # Note: The make command is required to generate cryptokit-1.7/src/stubs-md5.c earlier for substituteInPlace.
    make cryptokit-1.7/README.txt
```

Other than that it LGTM.

---

Considering <https://www.cvedetails.com/cve/CVE-2017-9772/>, I'm thinking I might have another check if we can use OCaml 4.07 with patches from master or something

---

Luckily that CVE shouldn't be an issue for us as from the description it should only apply to "binaries marked as setuid" - at least the official NixOS module should be fine (and no one else should setuid it). But other CVEs might apply and using a newer OCaml version seems like a good idea in general.

Alternatively we could also consider marking the package as insecure, using a newer version from the master branch, or asking upstream what they recommend / if they could e.g. tag a pre-release or something like that.

AFAIK sks's development is pretty much dead (or in the process - it's still being used though) and there are newer alternatives like https://gitlab.com/hagrid-keyserver/hagrid/.
Also: Feel free to add yourself as maintainer if you're actively using it (I'm not using it anymore so you could also remove me).

---

I'm currently debating with myself if we should maybe update to master, it has some compatibility fixes and even drops `camlp4` which is only available for <4.08

---

I think it's actually better to use current master (2021) than the latest release (2016).